src/DocScan/Session/Create/Task/RequestedTextExtractionTaskBuilder.php

```diff
@@ -12,38 +12,89 @@ class RequestedTextExtractionTaskBuilder
     private const ALWAYS = 'ALWAYS';
     private const FALLBACK = 'FALLBACK';
     private const NEVER = 'NEVER';
+    private const DESIRED = 'DESIRED';
+    private const IGNORE = 'IGNORE';
 
     /**
      * @var string
      */
     private $manualCheck;
 
-    public function withManualCheckAlways(): self
-    {
-        return $this->withManualCheck(self::ALWAYS);
-    }
+    /**
+     * @var string
+     */
+    private $chipData;
 
+    /**
+     * @param string $manualCheck
+     *
+     * @return $this
+     */
     public function withManualCheck(string $manualCheck): self
     {
         $this->manualCheck = $manualCheck;
         return $this;
     }
 
+    /**
+     * @return $this
+     */
+    public function withManualCheckAlways(): self
+    {
+        return $this->withManualCheck(self::ALWAYS);
+    }
+
+    /**
+     * @return $this
+     */
     public function withManualCheckFallback(): self
     {
         return $this->withManualCheck(self::FALLBACK);
     }
 
+    /**
+     * @return $this
+     */
```

Comment:
Is there a step we can add to pre-commit or Travis to ensure functions have type hints?

Reply:
We should be able to enforce this, but they're not critical as we now have return types on methods. We already check that these are correct when present, but they're not mandatory.